packages/gestalt/src/Masonry.js

```diff
@@ -145,7 +142,24 @@ export default class Masonry<T: { ... }> extends ReactComponent<Props<T>, State<
     this.setState({
       scrollTop: getScrollPos(scrollContainer),
     });
-  });
+  };
+
+  scrollTimeoutFn: ? TimeoutID = null;
+
+  // The browser scroll event API will fire LOTS of events. This waits until the events
+  // stop coming in to set the final scroll position
+  // $FlowFixMe[signature-verification-failure]
+  handleScroll = () => {
+    if (this.scrollTimeoutFn) {
+      clearTimeout(this.scrollTimeoutFn);
+    }
+    this.scrollTimeoutFn = setTimeout(this.updateScrollPosition, 100);
```

Comment:
100ms based on previous throttle value, happy to change.

Reply:
I'm not sure if it should change, but do you mind making it a top-of-file const with a descriptive name? If that is a knob we'll potentially want to tweak in the future, it'd be nice to make it easier to find.